src/Microsoft.Management.Configuration.Processor/Extensions/ValueSetExtensions.cs

```diff
@@ -90,5 +89,92 @@ public static IList<object> ToArray(this ValueSet valueSet)
 
             return sortedList.Values;
         }
+
+        /// <summary>
+        /// Performs a deep compare of the ValueSets.
+        /// </summary>
+        /// <param name="first">First ValueSet.</param>
+        /// <param name="second">Second ValueSet.</param>
+        /// <returns>Whether the two ValueSets equal.</returns>
+        public static bool ContentEquals(this ValueSet first, ValueSet second)
+        {
+            if (first.Count != second.Count)
+            {
+                return false;
+            }
+
+            foreach (var keyValuePair in first)
+            {
+                if (!second.ContainsKey(keyValuePair.Key))
+                {
+                    return false;
+                }
+
+                var firstValue = keyValuePair.Value;
+                var secondValue = second[keyValuePair.Key];
+
+                // Empty value check.
+                if (firstValue == null && secondValue == null)
+                {
+                    continue;
+                }
+                else if ((firstValue == null && secondValue != null) ||

```

Comment:
nit: could be simplified to:
```
else if (firstValue == null || secondValue == null)
```
since the first condition above already filters out the `&&` case.